pkg/virt-controller/watch/vm/vm.go

```diff
@@ -1621,10 +1621,34 @@ func syncVolumeMigration(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineIn
 	}
 	vmCond := controller.NewVirtualMachineConditionManager()
 	vmiCond := controller.NewVirtualMachineInstanceConditionManager()
-	// Something went wrong with the VMI while the volume migration was in progress
+
+	// Check if the volumes have been recovered and point to the original ones
+	srcMigVols := make(map[string]string)
+	for _, v := range vm.Status.VolumeUpdateState.VolumeMigrationState.MigratedVolumes {
+		if v.SourcePVCInfo != nil {
+			srcMigVols[v.VolumeName] = v.SourcePVCInfo.ClaimName
+		}
+	}
+	recoveredOldVMVolumes := true
+	for _, v := range vm.Spec.Template.Spec.Volumes {
+		name := storagetypes.PVCNameFromVirtVolume(&v)
+		origName, ok := srcMigVols[v.Name]
+		if !ok {
+			continue
+		}
+		if origName != name {
+			recoveredOldVMVolumes = false
+		}
+	}
+	if recoveredOldVMVolumes {
+		vm.Status.VolumeUpdateState.VolumeMigrationState.ManualRecoveryRequired = nil
```

Comment:
Unit test for this?

Reply:
I need to add entirely for the sycVolumeMigration function? Can we do it as a follow back?

Reply:
I don't see any coverage in unittests for `status.volumeUpdateState` but can be done in follow up

Reply:
I added the unit tests please, check the last commit